### PR TITLE
2.0.0 Release: Add API-Only Integration support

### PR DESCRIPTION
**Pull request by ChristianHansen**
Adds support for uploading images for API integrations and does some refactoring to surface response bodies when a non-2XX code is received from the Berbix API. This is the same code as the `2.x.x-beta` branch of the private fork.

After merging, I'll push a 2.0.0 git tag with the description of the breaking changes below and then `npm publish`.

Breaking changes in this release are:
 - Changes to the signature of the `request()` method expected for the `httpClient` passed to the `Client` constructor. See the README for the expected signature. Not passing an `httpClient` is still supported, and will continue to result in a default HTTP client being used by the SDK.
 - The some SDK methods can now throw an object with `status: number`, `error: string`, and `response: object` properties if a non-2XX status code is received from the API. Previously, the object thrown would depend on how the HTTP client handled non-2XX status codes, as objects thrown by the HTTP client would would be thrown by these methods if the default client was used, a `{status: number, error: string}` object would be thrown. The affected methods are
    - `refreshTokens()`
    - `createTransaction()`
    - `createApiOnlyTransaction()`
    - `createHostedTransaction()`
    - `fetchTransaction()`
    - `deleteTransaction()`
    - `updateTransaction()`
    - `overrideTransaction()`
    - `uploadImages()`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,6 +84,16 @@ Supported options:
 
 Creates a hosted transaction within Berbix to initialize the client SDK. This works the same as `createTransaction()` except that the object returned includes an explicit `hostedUrl` property for hosted transactions.
 
+##### `createApiOnlyTransaction(options: object): Tokens`
```
Comment (nwadams) [inline README.md:87]: Do we want to add support in CreateTransactionOptions for the selfie consent?

File: README.md [hunk 2]
```diff
@@ -145,6 +187,47 @@ The Unix timestamp in seconds at which the access and client tokens will expire.
...
 
+### `ImageUploadOpts`
+
+#### Properties
+
+##### `images: EncodedImage[]`
+
+Required. An array of `EncodedImage` objects representing images to upload.
+
+### `EncodedImage`
```
Comment (nwadams) [inline README.md:198]: Should we add support for the `SupplementaryData` and `ExtractedBarcode`

Comment (ChristianHansen) [reply README.md:198]: Going to kick this to another PR

**Review by nwadams — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (ChristianHansen) [reply README.md:87]: I think we should, but going to kick that to another PR for the sake of getting something usable released.

**Pull request merged**